File: MusicService.Entities/Entities/Music.cs
using MusicWebApp.Core.Abstract;

namespace MusicService.Entities.Entities
{
    public class Music : IEntity
    {
        public int Id { get; set; }
        public string? SingerName { get; set; }
        public string? Name { get; set; }
        public string? ImagePath { get; set; }
        public string? AudioPath { get; set; }
        public string AuthorId { get; set; }

        // Navigation properties 
        public virtual ICollection<Playlist>? Playlists { get; set; }
    }
}

File: musicwebapp.uiapp/src/types/Music.ts
import { Playlist } from "./Playlist";

export type Music = {
  id: number;
  singerName?: string | null;
  name?: string | null;
  imagePath?: string | null;
  audioPath?: string | null;
  authorId: string;
  playlists?: Playlist[] | null;
};
